src/applications/Machine.java

```diff
@@ -9,16 +9,72 @@
     private int totalWait; // total delay at this machine
     private int numTasks; // number of tasks processed on this machine
     private Job activeJob; // job currently active on this machine
+    private int machineNum; // to-do: make machines know their number
+    public static EventList eList;
 
-    // constructor
-    Machine() {
+    Machine(int machineNum){
+        this.machineNum = machineNum;
         jobQ = new LinkedQueue();
     }
 
-    public LinkedQueue getJobQ() {
-        return jobQ;
+    /**
+     * change the state of theMachine
+     *
+     * @return last job run on this machine
+     */
+    public Job changeState(int timeNow) {// Task on theMachine has finished,
+
+        // schedule next one.
+        Job lastJob = getActiveJob();
+        setActiveJob(timeNow);
+
+        return lastJob;
+    }
+
+    public void putJobOnQ(Job job){
+        jobQ.put(job);
+    }
+
+    public boolean hasNoJobs(){
+        return jobQ.isEmpty();
+    }
+
+    /*
+    If activeJob==null: sets activeJob to the next job on the Queue, unless the jobQ is empty then we change the "state"
+                        of the machine to be idling
+    Else: set the active job to be null, signifying that the machine is in its "change over state"
+
+    Also updates all relevant numbers/fields
+     */
+    private void setActiveJob(int timeNow){
```

Comment:
Good idea moving logic to here. It definitely makes the MachineShopSimulator more readable!

Reply:
I also really like this idea of moving this logic into the Machine class. By moving it here you fixed both a Data Class and Message Chain smell.